cli/oyster-cvm/src/commands/withdraw.rs

```diff
@@ -0,0 +1,225 @@
+use alloy::{
+    network::EthereumWallet,
+    primitives::{keccak256, Address, FixedBytes, U256},
+    providers::{Provider, ProviderBuilder},
+    signers::local::PrivateKeySigner,
+    sol,
+};
+use anyhow::{anyhow, Context, Result};
+use tracing::info;
+
+const ARBITRUM_ONE_RPC_URL: &str = "https://arb1.arbitrum.io/rpc";
+const OYSTER_MARKET_ADDRESS: &str = "0x9d95D61eA056721E358BC49fE995caBF3B86A34B"; // Mainnet Contract Address
+const MIN_WITHDRAW_AMOUNT: u64 = 1; // Minimum 0.000001 USDC
+const BUFFER_MINUTES: u64 = 7; // Required buffer time in minutes
+const SCALING_FACTOR: u128 = 1_000_000_000_000; // 1e12 scaling factor for contract values
+
+sol!(
+    #[allow(missing_docs)]
+    #[sol(rpc)]
+    OysterMarket,
+    "src/abis/oyster_market_abi.json"
+);
+
+pub async fn withdraw_from_job(
+    job_id: &str,
+    amount: Option<u64>,
+    max: bool,
+    wallet_private_key: &str,
+) -> Result<()> {
+    info!("Starting withdrawal process...");
+
+    // Setup wallet and provider with signer
+    let private_key = FixedBytes::<32>::from_slice(
+        &hex::decode(wallet_private_key).context("Failed to decode private key")?,
+    );
+    let signer = PrivateKeySigner::from_bytes(&private_key)
+        .context("Failed to create signer from private key")?;
+    let wallet = EthereumWallet::from(signer);
+
+    let provider = ProviderBuilder::new()
+        .with_recommended_fillers()
+        .wallet(wallet)
+        .on_http(
+            ARBITRUM_ONE_RPC_URL
+                .parse()
+                .context("Failed to parse RPC URL")?,
+        );
+
+    // Create contract instance
+    let market = OysterMarket::new(
+        OYSTER_MARKET_ADDRESS
+            .parse()
+            .context("Failed to parse market address")?,
+        provider.clone(),
+    );
+
+    let job_id_bytes = job_id.parse().context("Failed to parse job ID")?;
+
+    // Check if job exists and get current balance
+    let job = market
+        .jobs(job_id_bytes)
+        .call()
+        .await
+        .context("Failed to fetch job details")?;
+    if job.owner == Address::ZERO {
+        return Err(anyhow!("Job {} does not exist", job_id));
+    }
+
+    // Check if balance is zero
+    if job.balance == U256::ZERO {
+        return Err(anyhow!("Cannot withdraw: job balance is 0 USDC"));
+    }
+
+    // Try to settle the job first
+    info!("Attempting to settle job before withdrawal...");
```

Comment:
Every tx that is sent costs users ETH. So it probably isn't optimal to settle job before withdrawal, since it happens during withdrawal anyways.